### PR TITLE
add test cases for internal/data/db/scan.go

### PR DESCRIPTION
**Pull request by benjaminwilcox**
add test cases for internal/data/db/scan.go

### CODE REVIEW

File: internal/data/db/scan_test.go [hunk 1]
```diff
@@ -488,3 +557,122 @@ func setupSQLiteDB(t *testing.T) *gorm.DB {
...
+					High:        1,
+					Medium:      2,
+					Low:         3,
+					Info:        4,
+					Total:       10,
+				},
+			},
+			wantErr: false,
+		},
+		{
```
Comment (partkyle) [inline internal/data/db/scan_test.go:595]: These tests are strange because the code that's there is strange. It's not a reasonable expectation that `ctx` or `db` should be `nil` during normal operation. That's covered by using the `NewGormScanManager` method. That's the major reason to use constructors like that.

I think we need to remove all the nil checks instead. They are erroneous. If we checked nil on everything that could be nil, every time it could be nil, the entire codebase would be nil checks.

We need sufficient tests, and if there are `nil` panics they should get picked up by those tests. If not, we'll fix them. But these extra nil checks should be removed.

**Review by partkyle — CHANGES_REQUESTED** [review]

File: internal/data/db/scan_test.go (hunk 1 above)
Comment (benjaminwilcox) [reply internal/data/db/scan_test.go:551]: @partkyle the nil checks have been removed, are there any other changes?

**Review by dekkagaijin — APPROVED** [review]
